src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -560,12 +555,15 @@ private void WatchLaunchSettingsFile()
 
             try
             {
-                FileWatcher = new SimpleFileWatcher(_commonProjectServices.Project.GetProjectDirectory(),
-                                                    true,
-                                                    NotifyFilters.FileName | NotifyFilters.Size | NotifyFilters.LastWrite,
-                                                    LaunchSettingsFilename,
-                                                    LaunchSettingsFile_Changed,
-                                                    LaunchSettingsFile_Changed);
+                _ = JoinableFactory.RunAsync(async () =>
```

Comment:
There are two problems with discarding the task here, rather than blocking on it:

1. Any errors are not reported. We have a `Forget` extension method that will register any errors with the fault handler so we get data on them and can fix them.
2. It introduces a race condition between `Initialize` and `Dispose` in `LaunchSettingsProvider`. The `OnceInitializedOnceDisposed` base class will ensure that `Initialize`/`Dispose` won't overlap, but by spinning off some of the init work in an async way, it's possible that the dispose is called before `_launchSettingFileWatcher*` are initialized, which would then leak the watcher.

I think the race condition means that this needs to block, such as by using `Run` rather than `RunAsync`.